modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSourceTestCase.java

```diff
@@ -374,15 +378,16 @@ public void start() throws Exception {
       inOrder.verify(source).onStart(sourceCallback);
       return true;
     }));
+
     new PollingProber(TEST_TIMEOUT, TEST_POLL_DELAY).check(new JUnitLambdaProbe(() -> {
-      verifyLogMessage(infoMessages, "Message source 'source' on flow 'appleFlow' successfully started");
+      verifyLogMessage(logger.getMessages(), "Message source 'source' on flow 'appleFlow' successfully started");
       return true;
     }));
     checkNot(TEST_POLL_DELAY, TEST_TIMEOUT, () -> {
-      verifyLogMessage(infoMessages, "Message source 'source' on flow 'appleFlow' successfully reconnected");
+      verifyLogMessage(logger.getMessages(), "Message source 'source' on flow 'appleFlow' successfully reconnected");
       return true;
     });
-    setLogger(messageSource, LOGGER_FIELD_NAME, oldLogger);
+    logger.resetLevel();
```

Comment:
put this in an after of the test so that it will always be invoked even if an exception happens.